src/reducers-actions/orders.ts

```diff
@@ -38,21 +79,67 @@ export const reducer = (state: OrdersState, action: ReducerActionType): OrdersSt
         payload: { orders },
       } = action
 
-      // default sorting order is ascending by order id
-      // since we are starting from 0, reverse to have the latest on top
-      // make a copy first (slice(0)) to avoid mutating current state
-      const reversedOrders = orders.slice(0).reverse()
+      const reversedOrders = orders
+        // make a copy first (slice(0)) to avoid mutating current state
+        .slice(0)
+        // default sorting order is ascending by order id
+        // since we are starting from 0, reverse to have the latest on top
+        .reverse()
+        // remove deleted orders
+        .filter(order => !isOrderDeleted(order))
 
       return { ...state, orders: reversedOrders }
     }
+    case 'UPDATE_ORDERS': {
+      const {
+        payload: { orders: newOrders },
+      } = action
+      const { orders: currentOrders } = state
+
+      // Classify new orders between existing and deleted,
+      // keeping track of the ids and indexes
+      const { activeOrders, deletedOrdersIds } = classifyOrders(newOrders)
+
+      // Filter deleted orders and update existing orders
+      const filteredAndUpdatedOrders = currentOrders.reduce((acc: AuctionElement[], order) => {
+        if (activeOrders[order.id] !== undefined) {
+          // It's on activeOrders map? it has been updated.
+          // Get the updated from newOrders list by the index in the activeOrders map
+          acc.push(newOrders[activeOrders[order.id]])
+          // Remove from activeOrders map so we know it's not a new order
+          delete activeOrders[order.id]
+        } else if (!deletedOrdersIds.has(order.id)) {
+          // Not in the deletedOrdersIds set? it has not been modified.
+          // Keep the same
+          acc.push(order)
+        }
+        // Else? It has been deleted, do not include it
+        return acc
+      }, [])
+
+      // Anything left on activeOrders map is a new order
+      const reversedNewOrders = Object.values(activeOrders)
+        // Use their indexes to fetch from newOrders list
+        .map(index => newOrders[index])
+        // Reverse the result to have newest on top
+        .reverse()
+
+      // Add new orders first
+      const orders = reversedNewOrders.concat(filteredAndUpdatedOrders)
+
+      return { ...state, orders }
```

Comment:
This is kinda complicated. Wouldn't the following produce the same result?
```ts
   // put newest (by id) on top
  const reversedNewOrders = newOrders.slice(0).reverse()
  // keep track so that we don't duplicate orders
  const processedOrderIds = new Set<string>()

  // first we process reversedNewOrders
  // then currentOrders
  // but thanks to processedOrderIds newOrders override currentOrders with same id
  const updatedOrders = reversedNewOrders.concat(currentOrders).reduce<AuctionElement[]>((accum, order) => {
    // already included a potentially updated order
    // or the order was deleted
    if (processedOrderIds.has(order.id)) return accum

    // don't include deleted orders
    if (isOrderDeleted(order)) {
      // but consider them processed
      processedOrderIds.add(order.id)
      return accum
    }

    // this order is either one of the newOrders
    // or one of the currentOrders that wasn't overriden by newOrders
    // and it's not deleted
    accum.push(order)
    processedOrderIds.add(order.id)
    return accum
  }, [])

  return { ...state, orders: updatedOrders }
```

Reply:
Yes, I like your suggestion.
Much simpler, thanks!